src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs

```diff
@@ -460,7 +467,24 @@ private static bool StatementClosingDelimiterIsMissing(SyntaxNode currentNode)
         private static bool RequiredDelimiterIsMissing(SyntaxNode currentNode)
         {
             return currentNode.GetBrackets().closeBracket.IsMissing ||
-                currentNode.GetParentheses().closeParen.IsMissing;
+                currentNode.GetParentheses().closeParen.IsMissing ||
+                currentNode.GetBraces().closeBrace.IsMissing;
+        }
+
+        private static bool TryGetParentsWithBraces(SyntaxNode currentNode, [NotNullWhen(true)] out SyntaxNode? parentsWithBraces)
+        {
+            // Switch Expression could be followed by semicolon.
+            if (currentNode.GetAncestorsOrThis(IsSyntaxNodeWithBracesCouldBeFollowedBySemicolon).FirstOrDefault() is { } parent)
```

Comment:
The original reason is that I feel it might make sense to support
```
var x = switch bar
{
   "Case1" $$=> 1,
    _ => 3,
}
```
And move the caret to the end. 
But yeah this is too board, I have deleted this part.